src/components/files/FileActions/FileActionsMenu.tsx

```diff
@@ -158,13 +151,26 @@ export const FileActionsMenu: FC<FileActionsMenuProps> = (props) => {
       {...rest}
     >
       {menuItems.map((menuItem) => {
-        return menuItem.text === FileAction.NewVersion ? (
-          <div {...getRootProps()}>
-            <input {...getInputProps()} name="file-version-uploader" />
-            {renderedMenuItem(menuItem)}
-          </div>
-        ) : (
-          renderedMenuItem(menuItem)
+        const { text, directory, version } = menuItem;
+        return (item.type === 'Directory' && !directory) ||
+          (item.type === 'FileVersion' && !version) ? null : (
+          <MenuItem
+            key={text}
+            onClick={
+              text === FileAction.NewVersion
+                ? (event) => event.stopPropagation()
+                : (event) => handleActionClick(event, text)
+            }
+          >
+            {text === FileAction.NewVersion ? (
+              <span {...getRootProps()} className={classes.newVersionItem}>
+                <input {...getInputProps()} name="file-version-uploader" />
```

Comment:
Any other way I tried it, the click on the input would get swallowed by `event.stopPropagation()` above 😕

Reply:
Hmm ok. What are we stopping propagation for?

Reply:
The underlying row has a click listener that opens the file preview (if a `File`) or the directory (if a `Directory`)

Reply:
Ah. But we aren't interacting with that input directly. We just need the MUI button to be able to be clicked on.

Reply:
Correct, `react-dropzone` needs the input, but it abstracts the handling of it away

Reply:
What I mean is if we trigger the opening the native file picker ourselves then we don't need the dropzone to listen for that event. Which means it can be moved somewhere else and not clutter this list. Yes? Or am I missing something?